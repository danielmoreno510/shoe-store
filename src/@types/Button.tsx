type CustomButton = {
  color?: string;
  onPress(): any;
  text?: string;
  styleButton?: any;
  styleText?: any;
  children?: any;
};
