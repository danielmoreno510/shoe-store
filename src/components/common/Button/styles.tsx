import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  button: {
    height: 56,
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: 'white',
    fontSize: 20,
    fontWeight: 'bold',
  },
});
