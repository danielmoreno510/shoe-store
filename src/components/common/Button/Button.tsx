import React from 'react';
import {TouchableOpacity, Text} from 'react-native';

import styles from './styles';
import {primaryColor} from '../../../constants';

const CustomButton: React.FC<CustomButton> = ({
  color = primaryColor,
  onPress,
  text = null,
  styleButton = {},
  styleText = {},
  children = <></>,
}) => (
  <TouchableOpacity onPress={onPress} style={[styles.button, styleButton, {backgroundColor: color}]}>
    {text && <Text style={[styles.text, styleText]}>{text}</Text>}
    {children}
  </TouchableOpacity>
);

export default CustomButton;
