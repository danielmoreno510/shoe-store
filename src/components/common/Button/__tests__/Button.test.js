import React from 'react';
import {shallow} from 'enzyme';

import Button from '../Button';

describe('Button component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      primaryColor: '#002337',
      onPress: jest.fn(),
    };

    describe('WHEN the component is rendered without props', () => {
      const shallowWrapper = shallow(<Button {...testProps} />);

      it('THEN should display a regular button', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });

    describe('WHEN the component is rendered with props', () => {
      const customProps = {
        ...testProps,
        text: 'Test',
        disabled: true,
        styleButton: {color: 'red'},
        styleText: {color: 'red'},
      };
      const shallowWrapper = shallow(<Button {...customProps} />);

      it('THEN should display a regular button', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });
  });
});
