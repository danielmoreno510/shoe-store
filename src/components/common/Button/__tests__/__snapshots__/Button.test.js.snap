// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component GIVEN a set of valid props WHEN the component is rendered with props THEN should display a regular button 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 56,
        "justifyContent": "center",
      },
      Object {
        "color": "red",
      },
      Object {
        "backgroundColor": "#f1b30a",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Test
  </Text>
</ForwardRef>
`;

exports[`Button component GIVEN a set of valid props WHEN the component is rendered without props THEN should display a regular button 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 56,
        "justifyContent": "center",
      },
      Object {},
      Object {
        "backgroundColor": "#f1b30a",
      },
    ]
  }
/>
`;
